cli/src/commands/bookmark/list.rs

```diff
@@ -268,32 +268,22 @@ pub fn cmd_bookmark_list(
     if jj_lib::git::get_git_backend(repo.store()).is_ok() {
         // Print only one of these hints. It's not important to mention unexported
         // bookmarks, but user might wonder why deleted bookmarks are still listed.
-        let deleted_tracking = bookmark_list_items
-            .iter()
-            .filter(|item| item.primary.is_local() && item.primary.is_absent())
-            .map(|item| {
-                item.tracked.iter().any(|r| {
+        // There's no need to warn about pushing to the remote, because --deleted
+        // is now required for deleted bookmarks to be pushed.
+        let has_deleted_unexported = bookmark_list_items.iter().any(|item| {
+            item.primary.is_local()
+                && item.primary.is_absent()
+                && item.tracked.iter().any(|r| {
                     let remote = r.remote_name().expect("tracked ref should be remote");
-                    !jj_lib::git::is_special_git_remote(remote.as_ref())
+                    jj_lib::git::is_special_git_remote(remote.as_ref())
                 })
-            })
-            .max();
-        match deleted_tracking {
-            Some(true) => {
-                writeln!(
-                    ui.hint_default(),
-                    "Bookmarks marked as deleted will be *deleted permanently* on the remote on \
-                     the next `jj git push`. Use `jj bookmark forget` to prevent this."
-                )?;
```

Comment:
iirc, this is the hint for how/when scheduled deletion takes effect (and the deleted bookmarks disappear from the list.)

Maybe we'll need to suggest `jj git push --deleted`?

Reply:
tracked down the initial commit where this message was added: d01ecc5c46043351ea0a3ab982bb2fd606d81ea3

Reply:
> Why would we inform them about how to push deleted bookmarks in the bookmark list? 

because the user might not know why the deleted bookmark is still listed. As far as I understand it, it is the hint for how to "delete" deleted bookmarks from the list.

Reply:
Ok that does make sense. I put the hint back, just slightly changed so it makes sense with the current behavior.